File: grpc_api/domain/services/status_service.go
package services

import (
	"context"

	"github.com/janmbaco/CotizacionEspirituosa/grpc_api/domain/models"
	"github.com/janmbaco/CotizacionEspirituosa/grpc_api/domain/services/grpc_services"
)

type StatusService struct {
	grpc_services.UnimplementedStatusServiceServer
}

func (this *StatusService) GetStatus(context.Context, *models.StatusRequest) (*models.StatusResponse, error) {
	return &models.StatusResponse{Status: "OK"}, nil
}

File: web_app/src/app/providers/status.service.ts
import {Injectable} from '@angular/core';
import {StatusServiceClient} from '../../../domain/services/status_service_pb_service';
import {StatusRequest, StatusResponse} from '../../../domain/models/status_pb';
import { grpc } from '@improbable-eng/grpc-web';

@Injectable({
  providedIn: 'root'
})
export class StatusService {
  private client: StatusServiceClient;

  constructor() {

    this.client = new StatusServiceClient(
      'http://localhost:8081'
    );
  }

  getStatus(): Promise<string>{
    return new Promise<string>((resolve, reject) => this.client.getStatus(new StatusRequest(),
      (error, response: StatusResponse) => error == null ? resolve(response.getStatus()) : reject(error)));
  }
}
